synapse/handlers/message.py

```diff
@@ -904,32 +911,46 @@ def _send_dummy_events_to_fill_extremities(self):
             members = yield self.state.get_current_users_in_room(
                 room_id, latest_event_ids=latest_event_ids
             )
-
             user_id = None
+
             for member in members:
                 if self.hs.is_mine_id(member):
                     user_id = member
-                    break
-
-            if not user_id:
-                # We don't have a joined user.
-                # TODO: We should do something here to stop the room from
-                # appearing next time.
-                continue
-
-            requester = create_requester(user_id)
-
-            event, context = yield self.create_event(
-                requester,
-                {
-                    "type": "org.matrix.dummy_event",
-                    "content": {},
-                    "room_id": room_id,
-                    "sender": user_id,
-                },
-                prev_events_and_hashes=prev_events_and_hashes,
-            )
-
-            event.internal_metadata.proactively_send = False
-
-            yield self.send_nonmember_event(requester, event, context, ratelimit=False)
+                    requester = create_requester(user_id)
+                    try:
+                        event, context = yield self.create_event(
+                            requester,
+                            {
+                                "type": "org.matrix.dummy_event",
+                                "content": {},
+                                "room_id": room_id,
+                                "sender": user_id,
+                            },
+                            prev_events_and_hashes=prev_events_and_hashes,
+                        )
+
+                        event.internal_metadata.proactively_send = False
+
+                        yield self.send_nonmember_event(
+                            requester, event, context, ratelimit=False
+                        )
+                        break
+                    except ConsentNotGivenError:
+                        # Failed to send dummy event due to lack of consent, try another user
+                        user_id = None
+
+            if user_id is None:
```

Comment:
either write this as an `else` clause for the for loop, or use an explicit `succeeded` flag rather than abusing `user_id`

Reply:
succeed flag